runtime/service_protocol.cc

```diff
@@ -63,7 +63,7 @@ void ServiceProtocol::RemoveHandler(Handler* handler) {
 
 void ServiceProtocol::ToggleHooks(bool set) {
   for (const auto& endpoint : endpoints_) {
-    Dart_RegisterRootServiceRequestCallback(
+    Dart_RegisterIsolateServiceRequestCallback(
```

Comment:
@rmacnak-google: Can you confirm that this does not need to be called within an isolate scope? I seem to recall that you said it did when we last spoke.

Reply:
Yeah, I think I said that before, but Alex and I looked at the implementation today. The isolate service requests registered from the embedding API, unlike the ones registered from Dart code, are registered per VM instead of per isolate (though still dispatched to a particular isolate).